### ISSUE TITLE
Provide a network API

### ISSUE THREAD

**Issue by theclapp** [got-reload/got-reload#10]
Provide a network API so you can connect to a program and send new code to it.

**Comment by whereswaldon** [comment]
This seems handy, but is also (by definition) remote code execution. We shouldn't enable this by default, and we should think carefully about the implementation's access control mechanism.

**Comment by theclapp** [comment]
Indeed.  I agree on both counts.  I imagine a use-case of a) a private application or service in development, running on one's own machine, perhaps in a VM, that no one else can reach anyway, or b) full-on two-way TLS, with both client and server certificates required.  For something like this, self-signed would be fine, I should think, but obviously a "real" one would also be fine.

In any case, yeah, when and if we implement this, we can talk about the api then.  I just wanted to get the idea out there.
